nbsphinx_link/__init__.py

```diff
@@ -11,27 +11,95 @@
     "path": "relative/path/to/notebook"
 }
 
+Optionally the "extra-media" key can be added, if you notebook includes
+any media, i.e. images. The value can be an array or a string, with
+path/-s to the media file/-s or directory/-s.
+
 Further keys might be added in the future.
 """
 
-
+import errno
 import json
 import os
+import shutil
 
 from docutils import io, nodes, utils
 from docutils.utils.error_reporting import SafeString, ErrorString
 from nbsphinx import NotebookParser, NotebookError, _ipynbversion
 import nbformat
+from sphinx.util.logging import getLogger
 
 from ._version import __version__
 
 
+def copy_directory(src, dest):
+    """
+    Copies a directory or file from the path ``src`` to ``dest``.
+    Originally from: https://www.pythoncentral.io/how-to-recursively-copy-a-directory-folder-in-python/
```

Comment:
Thanks for the hint, I will keep things like that  in my mind in the future.
I put a link to the MIT License in the doc-string as stated on the [CC BY-NC 3.0 website](https://creativecommons.org/licenses/by-nc/3.0/deed.en_US).
I hope this is fine. 
IMO copying the whole text would clutter the doc string.

Reply:
The licensing term requires the entire text to be included (the CC guidelines do not affect the MIT license), but it doesn't need to be in the docstring, it can e.g. be in a comment.